tests/unit/test_charm.py

```diff
@@ -1,35 +1,88 @@
 # Copyright 2020 Ubuntu
 # See LICENSE file for licensing details.
 
+import hashlib
 import random
 import subprocess
 import unittest
+from base64 import b64decode
 from unittest.mock import Mock, call, mock_open, patch
 from uuid import uuid4
 
 from jinja2 import Template
 from ops.model import ActiveStatus, BlockedStatus
 from ops.testing import Harness
 
-from charm import NginxCharm
+from charm import NginxCharm, file_hash, install_ca_cert, write_file
 
-DEFAULT_CONFIG = {"host": str(uuid4()), "port": random.randint(10, 20)}
+DEFAULT_CONFIG = {
+    "host": str(uuid4()),
+    "port": random.randint(10, 20),
+    "ssl_cert": "dGVzdF9jZXJ0==",  # Padded base64 strings
+    "ssl_key": "dGVzdF9rZXk=",  # Padded base64 strings
+    "ca_cert": "dGVzdF9jYV9jZXJ0==",
+}
+NEW_CONFIG = {
+    "host": str(uuid4()),
+    "port": random.randint(10, 20),
+    "ssl_cert": "dGVzdF9jZXJ0==",  # Padded base64 strings
+    "ssl_key": "dGVzdF9rZXk=",  # Padded base64 strings
+    "ca_cert": "dGVzdF9jYV9jZXJ0==",
+}
 STORED_CONFIG = {"host": str(uuid4()), "port": random.randint(10, 20), "publishes": {}}
 
 
 class TestCharm(unittest.TestCase):
```

Comment:
suggestion: given how TLS is a new feature, I think it would be better to keep unit tests separate. You could have: a dedicated "enable TLS test" that verifies that if all three settings are configured the correct file-writing functions are called